### PR TITLE
Mobile 1293 Cordova urbanairship-accengage transition module

### PR DESCRIPTION
**Pull request by khmMouna**
### What do these changes do?
Creates a new cordova module that installs the transition module.

### How did you verify these changes?
Add plugin to your sample app project and verify from the log that Accengage is initiated. 

**Ps**: The first commit of this new repo needs to be reviewed also.

### CODE REVIEW

**Comment by BrianBatchelder** [comment]
Mouna, did the first commit in this new repo get reviewed?

**Comment by khmMouna** [comment]
@BrianBatchelder,  no it is no reviewed. The first commit is the default plugin.xml and package.json files generated by cordova. I will add a comment to this pull request.

File: package.json [hunk 1]
```diff
@@ -38,6 +38,5 @@
   "homepage": "https://github.com/urbanairship/urbanairship-accengage-cordova.git",
   "bugs": {
     "url": "https://github.com/urbanairship/urbanairship-accengage-cordova/issues"
-  },
-  "homepage": "https://github.com/urbanairship/urbanairship-accengage-cordova#readme"
+  }
```
Comment (BrianBatchelder) [inline package.json:42]: I suggest you add "accengage" to the "keywords" list in package.json

File: package.json [hunk 2]
```diff
@@ -38,6 +38,5 @@
   "homepage": "https://github.com/urbanairship/urbanairship-accengage-cordova.git",
```
Comment (BrianBatchelder) [inline package.json:39]: Could you please add a `README.md` to the repo?

File: plugin.xml [hunk 3]
```diff
@@ -26,6 +39,12 @@
         </config-file>
         <config-file parent="/*" target="AndroidManifest.xml">
         </config-file>
+
```
Comment (BrianBatchelder) [inline plugin.xml:60]: The "param" above should be something like:
```
<param name="android-package" value="com.urbanairship.accengage-cordova.UAirshipPlugin" />
```
Take a look at `plugin.xml` in the `urbanairship-cordova` repo for how our main Cordova package does it.

**Review by BrianBatchelder — CHANGES_REQUESTED** [review]
Looks good so far! Just a few minor requests.

File: package.json (hunk 1 above)
Comment (khmMouna) [reply package.json:42]: ok

File: plugin.xml (hunk 3 above)
Comment (khmMouna) [reply plugin.xml:60]: ok :)

File: README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Urban Airship Accengage Cordova Plugin
+
+This plugin supports Cordova apps running on both iOS and Android.
+
+### Issues
+
+Please visit http://support.urbanairship.com/ for any issues integrating or using this plugin.
+
+
```
Comment (BrianBatchelder) [inline README.md]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+### Requirements:
+ - cordova >= 9.0.0
+ - cordova-ios >= 5.0.1
+ - cococapods >= 1.7.3
```
Comment (BrianBatchelder) [inline README.md:12]: It also requires the urbanairship-cordova package, right?

File: plugin.xml [hunk 6]
```diff
@@ -10,22 +10,41 @@
...
+        
     </platform>
     
+    <!-- android -->
     <platform name="android">
         <config-file parent="/*" target="res/xml/config.xml">
-            <feature name="urbanairship-accengage-cordova">
-                <param name="android-package" value="urbanairship-accengage-cordova.urbanairship-accengage-cordova" />
+            <feature name="UAirshipPlugin">
+                <param name="android-package" value="com.urbanairship-accengage.cordova.UAirshipPlugin"/>
```
Comment (BrianBatchelder) [inline plugin.xml]:
```suggestion
                <param name="android-package" value="com.urbanairship.accengage-cordova.UAirshipPlugin"/>
```
It has to match whatever you put in the code, but it definitely should start with "com.urbanairship."

**Review by BrianBatchelder — CHANGES_REQUESTED** [review]
A few more things.

File: README.md (hunk 5 above)
Comment (khmMouna) [reply README.md:12]: yes, you are right

File: README.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+### Issues
+
+Please visit http://support.urbanairship.com/ for any issues integrating or using this plugin.
+
+### Requirements:
+ - cordova >= 9.0.0
+ - cordova-ios >= 5.0.1
+ - cococapods >= 1.7.3
+ - urbanairship-cordova >= 13.1.0
```
Comment (rlepinski) [inline README.md]: there is a way to make urbanairship-cordova a dependency in the plugin.xml

**Review by rlepinski — APPROVED** [review]

File: README.md (hunk 7 above)
Comment (khmMouna) [reply README.md]: Ok, I will add all the dependencies to the plugin.xml

File: plugin.xml [hunk 8]
```diff
@@ -10,22 +10,48 @@
     <keywords>cordova,urbanairship</keywords>
     <repo>https://github.com/urbanairship/urbanairship-accengage-cordova.git</repo>
     
+    <engines>
+        <engine name="cordova" version=">=9.0.0"/>
+        <engine name="cordova-ios" version=">=5.0.1"/>
```
Comment (BrianBatchelder) [inline plugin.xml:15]: Do you also need `cordova-android`?

File: plugin.xml [hunk 9]
```diff
@@ -10,22 +10,48 @@
     <keywords>cordova,urbanairship</keywords>
     <repo>https://github.com/urbanairship/urbanairship-accengage-cordova.git</repo>
     
+    <engines>
+        <engine name="cordova" version=">=9.0.0"/>
```
Comment (BrianBatchelder) [inline plugin.xml]: You should probably specify the same versions specified in our `urbanairship-cordova` `plugin.xml`

**Review by BrianBatchelder — COMMENTED** [review]
A couple of minor things.

File: plugin.xml [hunk 10]
```diff
@@ -10,22 +10,59 @@
     <keywords>cordova,urbanairship</keywords>
     <repo>https://github.com/urbanairship/urbanairship-accengage-cordova.git</repo>
     
+    <engines>
+        <engine name="cordova-android" version=">=4.1.0"/>
+        <engine name="cordova-ios" version=">=5.0.1"/>
+        <engine name="cordova" version=">=9.0.1"/>
+        <!-- <engine name="urbanairship-cordova" version=">=13.1.0"/> -->
```
Comment (khmMouna) [inline plugin.xml]: I commented this line for the moment until figuring out why I have this error when I try to add the `urbanairship-cordova` dependency.

Screen Shot 2020-02-25 at 10 58 33 AM

Comment (BrianBatchelder) [reply plugin.xml]: Does this work now with the correct version?

Comment (khmMouna) [reply plugin.xml]: Still not working even with the correct version :/

Comment (BrianBatchelder) [reply plugin.xml]: Version 10.1 has been released. Hopefully this will fix your issue.

Comment (khmMouna) [reply plugin.xml]: Finally it works 👍  but it's not related to the new version. The problem is in the engine tag, it misses some required attributes for custom frameworks (platform and scriptSrc). The format of custom framework should be like this : 
`<engines>`
`<engine name="my_custom_framework" version="1.0.0" platform="ios|android" scriptSrc="path_to_my_custom_framework_version"/>`
 `</engines>`

**Review by BrianBatchelder — APPROVED** [review]
👍

**Review by BrianBatchelder — APPROVED** [review]
👍

**Pull request merged**